gen_future_weather_file.py

```diff
@@ -0,0 +1,425 @@
+
+    # Write the generated string to the passed file.
+    # We pre-generate the string as it is much quicker to append to a
+    # string than it is to write to a file.
+    with open(filename, "w+") as epw:
+        epw.write(epw_file)
+
+
+def get_daily_averages(epw_data: list):
+    """ get_daily_averages(list)
+
+        Calculates each day's average from the passed epw data, and returns
+        a list of those averages.
+
+        Args:
+            epw_data(list): The data read from the epw that we will be
+                    averaging with.
+
+        Returns:
+            A list of averaged data.
+    """
+
+    averages = [[] for _ in range(366)]
+    running_averages = [0 for _ in range(len(epw_data[0]))]
+    current_day = epw_data[0][0]
+
+    for data_row in epw_data:
+
+        if data_row[0].day != current_day.day:
+            for i in range(len(running_averages)):
+                running_averages[i] /= 24
+
+            averages[current_day.timetuple().tm_yday - 1] = running_averages[:]
+            running_averages = [0 for _ in range(len(epw_data[0]))]
+
+            current_day = data_row[0]
+
+        for index, datapoint in enumerate(data_row):
+            if type(datapoint) is float or type(datapoint) is int:
+                running_averages[index] += datapoint
+
+    return averages
+
+
+def get_climate_data(climate_file: str,
+                     lat: float,
+                     long: float,
+                     cdfvariable: str,
+                     time_range: list
+                     ):
+    """ get_climate_data(str, float, float, list)
+
+        Gets a list of data for each day of each year within time_range from
+        the climate file where the location is closest to lat, long.
+
+        Args:
+            climate_file(str): The path to the climate file to read from.
+            lat(float): The latitude to read data from.
+            long(float): The longitude to read data from.
+            cdfvariable(str): The variable to read from the netcdf file.
+            time_range(range): The range of years to read data from, to.
+
+        Returns:
+            a list with 366 lists. One for each day of the year in a leap year.
+            Each list contains all the data found in the file for that
+            specific day, with each entry in the inner list is 1 year's entry
+            for that day.
+    """
+
+    # Read the climate data out of the climate file.
+    raw_data = Dataset(climate_file)
+
+    # Get a list of the dates in the climate file.
+    data_dates = cdf.num2date(raw_data["time"][:], raw_data["time"].units)
+
+    # Get the latitude of each location in the file.
+    lat_data = raw_data.variables["lat"][:]
+
+    # Get the logitude of each location in the file.
+    long_data = raw_data.variables["lon"][:]
+
+    # Find the incides of the data with the closest lat and long to
+    # those passed.
+    lat_index = lat_data.tolist().index(
```

Comment:
`.argmin()` returns the index of the minimum value, so couldn't this whole thing simply be:

~~~python
lat_index = np.abs(lat_data - lat).argmin()`
~~~
?